bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -554,10 +555,12 @@ def from_api_repr(cls, resource):
         instance = cls(name)
         types = instance.struct_types
         for item in resource['parameterType']['structTypes']:
-            types[item['name']] = item['type']
+            types[item['name']] = item['type']['type']
```

Comment:
I suspect this client library might have problems if someone tries to put an array or struct into a struct, but I won't block this PR for that. (I didn't write a sample for that use case.)